common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -200,6 +207,25 @@ def get_html(self):
                 val_profiles = ["youtube", "desktop_webm", "desktop_mp4"]
                 val_video_urls = edxval_api.get_urls_for_profiles(self.edx_video_id, val_profiles)
 
+                # download youtube transcript for html5 video
+                if not self.sub and val_video_urls["youtube"]:
+                    #cache the transcript request
+                    cache_key = 'html5_' + val_video_urls["youtube"]
+                    html5_transcript_cache = cache.get(cache_key)
+                    if not html5_transcript_cache:
+                        try:
+                            download_youtube_subs(val_video_urls["youtube"], self, settings)
+                            item = self.runtime.modulestore.get_item(self.location)
+                            youtube_sub = val_video_urls["youtube"]
+                            item.sub = youtube_sub
+                            user = User.objects.get(id=self.system.user_id)
+                            item.save_with_metadata(user)
+                            transcript_found = True
+                        except GetTranscriptsFromYouTubeException:
+                            transcript_found = False
+
+                        cache.set(cache_key, {'youtube_hit': True, 'found': transcript_found})
```

Comment:
@adampalay ```self.sub``` is not instance of ```VideoDescriptor``` and we cannot perform the update operations without ```VideoDescriptor``` instance so thats why i first get ```VideoDescriptor``` instance using location and update the it using ```item.sub```